packages/expo/cli/api/getExpoSchema.ts

```diff
@@ -0,0 +1,86 @@
+import { JSONObject } from '@expo/json-file';
+import fs from 'fs';
+import schemaDerefSync from 'json-schema-deref-sync';
+import path from 'path';
+
+import { EXPO_UNIVERSE_DIR } from '../utils/env';
+import { CommandError } from '../utils/errors';
+import { createCachedFetch } from './rest/client';
+
+export type Schema = any;
+
+export type AssetSchema = {
+  fieldPath: string;
+};
+
+const schemaJson: { [sdkVersion: string]: Schema } = {};
+
+// TODO: Maybe move json-schema-deref-sync out of api (1.58MB -- lodash)
+// https://packagephobia.com/result?p=json-schema-deref-sync
+async function getSchemaAsync(sdkVersion: string): Promise<Schema> {
+  const json = await getSchemaJSONAsync(sdkVersion);
+  return schemaDerefSync(json.schema);
+}
+
+/**
+ * Array of schema nodes that refer to assets along with their field path (eg. 'notification.icon')
+ *
+ * @param sdkVersion
+ */
+export async function getAssetSchemasAsync(sdkVersion: string | undefined): Promise<string[]> {
+  // If no SDK version is available then fall back to unversioned
+  const schema = await getSchemaAsync(sdkVersion ?? 'UNVERSIONED');
+  const assetSchemas: string[] = [];
+  const visit = (node: Schema, fieldPath: string) => {
+    if (node.meta && node.meta.asset) {
+      assetSchemas.push(fieldPath);
+    }
+    const properties = node.properties;
+    if (properties) {
+      Object.keys(properties).forEach((property) =>
+        visit(properties[property], `${fieldPath}${fieldPath.length > 0 ? '.' : ''}${property}`)
+      );
+    }
+  };
+  visit(schema, '');
+
+  return assetSchemas;
+}
+
+async function getSchemaJSONAsync(sdkVersion: string): Promise<{ schema: Schema }> {
+  if (EXPO_UNIVERSE_DIR) {
+    return JSON.parse(
+      fs
+        .readFileSync(
+          path.join(EXPO_UNIVERSE_DIR, 'server', 'www', 'xdl-schemas', 'UNVERSIONED-schema.json')
+        )
+        .toString()
+    );
+  }
```

Comment:
I never use it but I also don't work on universe very much. I'd love to drop this.

Reply:
This is just for schema warning validation so it's not something I've ever tested. Since we test against an older version we know that assembling and warning always works.